src/modules/auth/sso/index.ts

```diff
@@ -0,0 +1,54 @@
+import { chromium } from 'playwright';
+import { DASHLANE_APP_REGEX, extractSsoInfoFromUrl } from './utils';
+import { performSSOVerification } from '../../../endpoints/performSSOVerification';
+
+interface SSOParams {
+    requestedLogin: string;
+    serviceProviderURL: string;
+}
+
+const openIdPAndWaitForRedirectURL = async (serviceProviderURL: string): Promise<URL> => {
+    return new Promise((resolve, reject) => {
+        void (async () => {
+            try {
+                const browser = await chromium.launch({ headless: false, channel: 'chrome' });
+                const context = await browser.newContext();
+                const page = await context.newPage();
+
+                page.on('framenavigated', (frame) => {
+                    const url = page.url();
+                    if (frame === page.mainFrame() && url.match(DASHLANE_APP_REGEX)) {
+                        void browser.close();
+                        resolve(new URL(url));
+                    }
+                });
+
+                browser.on('disconnected', reject);
+
+                await page.goto(serviceProviderURL);
+            } catch (error) {
+                reject(error);
+            }
+        })();
+    });
+};
+
+export const doSSOVerification = async ({ requestedLogin, serviceProviderURL }: SSOParams) => {
+    const redirectURL = await openIdPAndWaitForRedirectURL(serviceProviderURL);
```

Comment:
So this will wait till end of time if there's no match.

Reply:
that's a valid comment, I'll probably wait for the tab to be closed or a long timeout